### DISCUSSION TITLE
[Warning] Failed to load slave replication state from table mysql.gtid_slave_pos

### DISCUSSION THREAD

**Discussion by alpha754293** [photoprism/photoprism#4657] [Q&A]
This is a follow-on to the other discussion topic where my database got *very* corrupted. After my other discussion thread was posted, I was able to use `mariadb-dump` to backup the database, but when I exited out of the mariadb container, stopped said container, and then tried to restart it again, it said that I needed to put the `--innodb-force-recovery` to `5` to be able to get back into it. So I did that, but then I couldn't do a `DROP DATABASE photoprism` so I had to change `--innodb-force-recovery` to 4 rather than 5. 

I was then able to DROP the database, stop mariadba/mysql, ran `rm -rf -i ib*` from `/var/lib/mysql` and then recreated the DATABASE, and then ran the import.

I got this error after importing the database from doing a `mariadb-dump`:

```
[Warning] failed to load slave replication state from table mysql.gtid_slave_pos: 1932: Table `mysql.gtid_slave_pos` doesn't exist in engine
```

I'm not really sure what happened or why this instance of Photoprism got so badly corrupted, but I am going to try deleting the entire database and start over from scratch.

Just wanted to report the issue, to see if there is a better, recommend solution to fix this issue so that if it happens to me again, hopefully there is a better way to recover from this kind of an error other than rebuilding the entire database from scratch again.

Thank you.

**Comment by omerdduran** [comment]
Hi. By default, [PhotoPrism 240523-923ee0cf7](https://docs.photoprism.app/release-notes/#may-23-2024) and newer versions automatically create daily database backups for you, with up to 3 copies being retained. You can restore the database from these backups. However, remember that automatic backups are not full backups. The easiest way to create a full backup is to first run the backup command to create a database dump. Then back up your originals and storage folders using any standard file backup utility.

https://docs.photoprism.app/user-guide/backups/
https://docs.photoprism.app/user-guide/backups/restore/
